File: backend/app/schemas/pricing.py
"""Schemas für die Preis-Indikation."""

from pydantic import BaseModel, Field


class PriceEstimateIn(BaseModel):
    name: str = Field(min_length=1, max_length=300)
    description: str | None = None


class PriceEstimateOut(BaseModel):
    price: float | None
    currency: str
    sources: list[str] = []
    mode: str
    note: str | None = None


class PricingStatus(BaseModel):
    available: bool
    mode: str
    backend: str | None = None
    search_provider: str = "none"

File: frontend/src/services/pricing.ts
import api from './api'

export interface PriceEstimate {
  price: number | null
  currency: string
  sources: string[]
  mode: string
  note: string | null
}

export interface PricingStatus {
  available: boolean
  mode: string
  backend: string | null
}

export const pricingStatus = () =>
  api.get<PricingStatus>('/api/pricing/status').then((r) => r.data)

export const estimatePrice = (name: string, description?: string | null) =>
  api
    .post<PriceEstimate>('/api/pricing/estimate', { name, description })
    .then((r) => r.data)
